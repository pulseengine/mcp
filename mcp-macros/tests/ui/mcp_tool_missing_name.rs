//! Test case that should fail to compile due to missing name attribute in mcp_server

use pulseengine_mcp_macros::{mcp_tools, mcp_server};

// This should fail to compile because name is required for mcp_server
#[mcp_server(description = "A server without a name")]
#[derive(Clone, Default)]
struct ServerWithoutName;

#[mcp_tools]
impl ServerWithoutName {
    pub fn some_tool(&self) -> String {
        "This should not compile".to_string()
    }
}

fn main() {}