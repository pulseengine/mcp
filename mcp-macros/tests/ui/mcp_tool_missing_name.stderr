error: Server name is required. Use #[mcp_server(name = "Your Server Name")]
 --> tests/ui/mcp_tool_missing_name.rs:6:1
  |
6 | #[mcp_server(description = "A server without a name")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `mcp_server` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `ServerWithoutName` in this scope
  --> tests/ui/mcp_tool_missing_name.rs:11:6
   |
11 | impl ServerWithoutName {
   |      ^^^^^^^^^^^^^^^^^ not found in this scope
